### PR TITLE
doc: vsc: update modifying application page

### PR DESCRIPTION
**Pull request by greg-fer**
Added information about VSC extension options to Modifying application.
Updated other pages accordingly.
NCSDK-14998.

Signed-off-by: Grzegorz Ferenc <Grzegorz.Ferenc@nordicsemi.no>

### CODE REVIEW

**Comment by greg-fer** [comment]
Tried to avoid lines touched by #7484

File: doc/nrf/gs_testing.rst
Comment (kittydepa) [inline doc/nrf/gs_testing.rst:22]: I believe this should be:
`The |VSC| extension pack ... `

Is this right @Avalei, or is it fine as it?

**Review by kittydepa — COMMENTED** [review]
LGTM, just waiting for a second opinion on my previous comment, then I will approve :)

**Review by FrancescoSer — APPROVED** [review]

File: doc/nrf/gs_testing.rst
Comment (Avalei) [reply doc/nrf/gs_testing.rst:22]: In this particular case let's add "extension pack" onto the end because the next sentence mentions nRF Terminal, so this could be a bit confusing to users.

Comment (Avalei) [inline doc/nrf/gs_testing.rst:22]:
```suggestion
The |VSC| extension pack is a complete IDE for developing applications for nRF91, nRF53 and nRF52 Series devices.
```

File: doc/nrf/gs_modifying.rst
Comment (Avalei) [inline doc/nrf/gs_modifying.rst:227]:
```suggestion
This extension organizes the Kconfig options in a hierarchical list and lets you select the desired options.
```

File: doc/nrf/gs_modifying.rst
Comment (Avalei) [inline doc/nrf/gs_modifying.rst:228]: We have three different save options actually. Are we assuming here that the user is selecting their kconfig options for the first time and needs to just save them to a new .conf file?

File: doc/nrf/gs_modifying.rst
Comment (Avalei) [inline doc/nrf/gs_modifying.rst:238]: The "Jump to Item" options is only visible on options you have already enabled and which are showing up in the Changes tab. If you want to search for a new kconfig option to enable, it's better to type in the name of the option in the search field.

File: doc/nrf/gs_modifying.rst
Comment (Avalei) [inline doc/nrf/gs_modifying.rst:264]: Is "permanently" the right word to use here? I believe both "Save to file" and "Save (minimal)" are both technically permanent saves it just depends on if there is an existing .conf file already or not. @trond-snekvik is that right? We might want to provide a little more context around the saving options for this section.

- Save: saves temporary changes to the <build>/zephyr/.conf file.
- Save to file: saves changes to an existing .conf file.
- Save (minimal): saves the smallest subset of settings to a new .conf file.

File: doc/nrf/gs_modifying.rst
Comment (Avalei) [inline doc/nrf/gs_modifying.rst:287]:
```suggestion
See `Building an application`_ in the |VSC| documentation.
```

**Review by Avalei — CHANGES_REQUESTED** [review]

**Review by divipillai — APPROVED** [review]

File: doc/nrf/gs_modifying.rst
Comment (trond-snekvik) [inline doc/nrf/gs_modifying.rst:229]: This whole paragraph should be merged with the previous one about menuconfig, I think. This is describing "menuconfig", but for VS Code. The nRF Kconfig extension does a lot more than this Kconfig menu interface, so I think we should be referring to this feature as the VS Code Kconfig menu (or whatever we ended up calling it, @Avalei )

File: doc/nrf/gs_modifying.rst
Comment (trond-snekvik) [reply doc/nrf/gs_modifying.rst:264]: I would say "permanently" is the right concept, but I don't know if it's the right phrase. It's definitely "non-temporary" at least 😄 Agree with the rest!

File: doc/nrf/gs_testing.rst
Comment (greg-fer) [reply doc/nrf/gs_testing.rst:22]: Added `pack` to the next sentence after this one, because IDE is always referred to as "VSC extension", not "VSC extension pack". So `pack` makes sense in the next sentence next to nRF Terminal.

File: doc/nrf/gs_modifying.rst
Comment (greg-fer) [reply doc/nrf/gs_modifying.rst:228]: I'm using the differentiation mentioned in the VSC docs. As we're talking about temporary changes, I'm mentioning "Save". In line 264, where we mention permanent changes, I mention "Save to file".

File: doc/nrf/gs_modifying.rst
Comment (greg-fer) [reply doc/nrf/gs_modifying.rst:238]: Edited. Please check.

File: doc/nrf/gs_modifying.rst
Comment (greg-fer) [reply doc/nrf/gs_modifying.rst:264]: Hm, on second thoughts, maybe I don't need to describe the UI button here, so I rephrased for simplicity. But I did keep the "Save" in line 228.

**Comment by greg-fer** [comment]
@Avalei , @trond-snekvik , @kittydepa , addressed your edits. Waiting for #7484 for the final rebase, but you could approve already. Worst case, this goes in before and Divya's PR removes SES sections that I kept.

File: doc/nrf/gs_modifying.rst
Comment (Avalei) [inline doc/nrf/gs_modifying.rst:223]:
```suggestion
While it is possible to edit the :file:`.config` file directly, you should use the nRF Kconfig GUI in |VSC| or a tool like menuconfig or guiconfig to update it.
```

File: doc/nrf/gs_modifying.rst
Comment (Avalei) [inline doc/nrf/gs_modifying.rst:227]:
```suggestion
To save the changes made using the nRF Kconfig GUI, click the :guilabel:`Save` button.
```

File: doc/nrf/gs_modifying.rst
Comment (Avalei) [reply doc/nrf/gs_modifying.rst:228]: In a conversation on May 9th we decided that we will use "_nRF Kconfig when referring to the extension as a whole, and "the nRF Kconfig GUI" when talking about the interface itself._"

File: doc/nrf/gs_modifying.rst
Comment (Avalei) [inline doc/nrf/gs_modifying.rst:237]:
```suggestion
To locate a specific configuration option, use the filter (:guilabel:`Search modules` field in the nRF Kconfig GUI or :guilabel:`Jump to` in menuconfig and guiconfig).
```

**Review by Avalei — CHANGES_REQUESTED** [review]

File: doc/nrf/gs_modifying.rst
Comment (greg-fer) [reply doc/nrf/gs_modifying.rst]: Edits applied. Please re-check.

**Review by Avalei — APPROVED** [review]

**Review by kittydepa — APPROVED** [review]

**Review by trond-snekvik — APPROVED** [review]

**Pull request merged**
